app/crud/common.py

```diff
@@ -0,0 +1,63 @@
+from pydantic import ValidationError
+from .labels import Label, PredefinedLabel
+from .errors import summarize_error
+
+## TODO
+# migrate - maybe we don't move out of sql?
+# copy predefined labels to labels - maybe we dont move out of sql?
+# same again for probes
+# hook up to stored procs for (CRUD and validation)
+# more tests
+_TYPES = {"LABEL": Label, "PREDEFINED_LABEL": PredefinedLabel}
+
+
+def create_entity(session, entity_type, entity):
```

Comment:
You'll notice that the begin/commit-rollback is no longer here.

Long-form answer: I pushed it down into the BaseModel class. The original patch had the entire CRUD call (including updating the view after the fact) wrapped in the transaction. This fails when we try to re-create the view after doing the CRUD work. Pushing it down lets us do all of the work for CRUD, commit, and then update the view.

This also has the benefit of converging the streamlit code path with the quickly incoming ADMIN procedures path.

Reply:
dont mind moving it inside but it seems we are re-using the logic and that could still live in a context manager

Reply:
Got it.